packages/components/data-resource/data-resource-create-form.tsx

```diff
@@ -94,13 +95,20 @@ const isFormValid = (form: CreateFormDataState) => {
     pvc,
     pvcFolderName,
   } = form;
+  const secretValid = secretName || (secretKey && accessKey);
+  const awsValid =
+    region && secretValid && target && provider === BC_PROVIDERS.AWS;
+  const ibmValid =
+    endpoint && secretValid && target && provider === BC_PROVIDERS.IBM;
+  const s3Valid =
+    endpoint && secretValid && target && provider === BC_PROVIDERS.S3;
+  const azureValid = secretValid && target && provider === BC_PROVIDERS.AZURE;
+  const fileSystemValid =
+    pvc && pvcFolderName && provider === BC_PROVIDERS.FILESYSTEM;
```

Comment:
Wouldn't this be less intensive? 
```suggestion
  const secretValid = secretName || (secretKey && accessKey);
const isBCproviderValid = false;
  if(provider === BC_PROVIDERS.AWS) {
  const isBCproviderValid =
    region && secretValid && target && provider;
  } else if () {
  ...
  }
 ```

Reply:
we can do this but it will add a bunch of if-else statements. instead, since we need only boolean values, we can short circuit the condition by checking if the provider is not matched.

Reply:
@SanjalKatiyar any opinion on this?

Reply:
add a Switch case...
based upon which provider is selected we will return true/false...